### ISSUE TITLE
ObservationWebFilterChainDecorator cancel signal triggers Observation::start

### ISSUE THREAD

**Issue by oemergenc** [spring-projects/spring-security#13827]
**Describe the bug**
The ObservationWebFilterChainDecorator should not call start in an onCancel trigger when wrapping a WebFilterChain, see here:
https://github.com/spring-projects/spring-security/blob/f91a51bab70d3f2105ab814576e22cf7a8c421e2/web/src/main/java/org/springframework/security/web/server/ObservationWebFilterChainDecorator.java#L343C11-L343C11

This issue was also discussed here:
https://github.com/micrometer-metrics/tracing/issues/357

**Comment by Hiosdra** [comment]
Hi, any progress with this?

**Comment by jzheaux** [comment]
Thanks, @Hiosdra, for reporting this issue. It should be resolved in #14031.

**Comment by janekbettinger** [comment]
Hi @jzheaux , is it on purpose that [`SimpleAroundWebFilterObservation#wrap(WebFilterChain)`](https://github.com/spring-projects/spring-security/blob/202d0fccbf7310117b9c36425e861805ef16b3f6/web/src/main/java/org/springframework/security/web/server/ObservationWebFilterChainDecorator.java#L350C1-L354C17) has swapped semantics compared to the corresponding method [`SimpleWebFilterObservation#wrap(WebFilterChain)`](https://github.com/spring-projects/spring-security/blob/202d0fccbf7310117b9c36425e861805ef16b3f6/web/src/main/java/org/springframework/security/web/server/ObservationWebFilterChainDecorator.java#L552)?
The first method triggers a _start_ of the "after observation" even if a _cancel_ or _error_ signal is received. This might be intended, however, I'm currently investigating an issue where replacing `start()` with `close()` seems to help.

The behavior we sporadically observe looks quite similar to https://github.com/micrometer-metrics/tracing/issues/357 with Spring Boot 3.2.1 (`spring-security-web:6.2.1`) and WebFlux (`reactor-core:3.6.1`):

```
java.lang.IllegalStateException: Span wasn't started - an observation must be started (not only created)
	at io.micrometer.tracing.handler.TracingObservationHandler.getRequiredSpan(TracingObservationHandler.java:209)
	Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException: 
Error has been observed at the following site(s):
	*__checkpoint ⇢ ServerWebExchangeReactorContextWebFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ org.springframework.security.web.server.WebFilterChainProxy [DefaultWebFilterChain]
	*__checkpoint ⇢ HTTP GET "/an-endpoint" [ExceptionHandlingWebHandler]
Original Stack Trace:
	at io.micrometer.tracing.handler.TracingObservationHandler.getRequiredSpan(TracingObservationHandler.java:209)
	at io.micrometer.tracing.handler.DefaultTracingObservationHandler.onStop(DefaultTracingObservationHandler.java:50)
	at io.micrometer.observation.ObservationHandler$FirstMatchingCompositeObservationHandler.onStop(ObservationHandler.java:197)
	at io.micrometer.observation.SimpleObservation.lambda$notifyOnObservationStopped$0(SimpleObservation.java:268)
	at java.base/java.util.ArrayDeque$DescendingIterator.forEachRemaining(Unknown Source)
	at io.micrometer.observation.SimpleObservation.notifyOnObservationStopped(SimpleObservation.java:268)
	at io.micrometer.observation.SimpleObservation.stop(SimpleObservation.java:188)
	at org.springframework.security.web.server.ObservationWebFilterChainDecorator$PhasedObservation.close(ObservationWebFilterChainDecorator.java:763)
	at org.springframework.security.web.server.ObservationWebFilterChainDecorator$AroundWebFilterObservation$SimpleAroundWebFilterObservation.close(ObservationWebFilterChainDecorator.java:301)
	at org.springframework.security.web.server.ObservationWebFilterChainDecorator$AroundWebFilterObservation$SimpleAroundWebFilterObservation.lambda$wrap$3(ObservationWebFilterChainDecorator.java:366)
```

It fails to _close_ the apparently unstarted "after observation" (`this.after.close();`) as it tries to _stop_ it first. This should not happen as the `close` method stops only previously _started_ observations. There seems to be a mismatch between `this.phase` (`0 = initial / 1 = started / 2 = stopped / 3 = closed`) and the actual state of the `this.after` observation. Besides of the code fragment mentioned in my question above, I haven't found anything suspicious, yet.
